rfcs/001 - EC2 schema.md

```diff
@@ -0,0 +1,136 @@
+
+The definition of a solid base now will imply that new features added to this module will not translate into breaking changes to the model.
+
+Taking a look at the AWS EC2 console options, we would still need to add the following to reach EC2 module completeness:
+
+![AWS Console EC2](./assets/001/aws_console_ec2.png)
+
+- Instance types with Availability zone relationship
+- Launch templates
+- Spot instances
+- Dedicated hosts
+- Saving plans
+- Reserved instances
+- Scheduled instances
+- Capacity reservations
+- Auto Scaling Groups
+- AMIs management
+- EBS volume snapshots and life cycle
+- Placement groups
+- Key Pairs
+- Network interfaces
+
+The `instance` table still needs more advanced options to be defined. Some of those missing properties are: instance auto-recovery, shutdown behaviour, stop - hibernate behaviour, termination protection, stop protection, detailed cloudWatch monitoring, elastic GPU, credit specification, tenancy, RAM disk ID, kernel ID or Metadata accessible (I might be missing others). Additionally, some of the features listed above will need to be added to the `instance` table as FKs.
+
+The current EC2 model looks like this:
+
+![Current EC2 model](./assets/001/current_schema.png)
+Note: Could be also accessed via https://dbdocs.io/alejandro/iasql
+
+## Proposal
+
+After considering alternatives like splitting everything into more modules or having different `instance` tables around based on what service creates them, the proposal is to keep the schema growing in this same module. The relationships, though, will be added where they belong and rely on the instance metadata for the no needed ones for the CRUD of the `instance` table.
+
+We will review the needed features, see their implications in the model and get a final version of the ECS schema.
+
+- Creating an instance using the UI/API lets you add some advanced configuration that usually is not used but is there. The screenshot below shows part of the extra configuration. At least the following columns will need to be added to the `instance` table: instance auto-recovery, shutdown behaviour, stop - hibernate behaviour, termination protection, stop protection, detailed cloudWatch monitoring, elastic GPU, credit specification, tenancy, RAM disk ID, kernel ID or Metadata accessible.
+
+![EC2 instance advance properties](./assets/001/ec2_advance_properties.png)
+
+- Instance types depend on availability zones. We need to create a table relating them using a composite key. We also need to add the `availability_zone` column to the `instance` table to relate with the new `instance_type` table using the composite key.
```

Comment:
Once/if this is accepted, we should create a task and tackle this one soon-ish?

Reply:
yes, this would be easy to do